src/OpenFeature.Contrib.Providers.Flagd/FlagdProvider.cs

```diff
@@ -105,6 +111,21 @@ public static string GetProviderName()
         /// </summary>
         internal Resolver.Resolver GetResolver() => _resolver;
 
+        /// <inheritdoc/>
+        public override Task Initialize(EvaluationContext context)
+        {
+            return Task.Run(() =>
+            {
+                this._resolver.Init();
+                this._status = ProviderStatus.Ready;
+
+            }).ContinueWith((t) =>
+            {
+                this._status = ProviderStatus.Error;
+                if (t.IsFaulted) throw t.Exception;
+            });
+        }
+
```

Comment:
@bacherfl I had to move the init from the constructor to the proper init function as per the spec. The flagd provider was written before we had proper init/shutdown so it hand't yet taking advantage of this.